### PR TITLE
(chore) Bump i18next, react-i18next and i18next-parser and support for pluralization

### PR DESCRIPTION
**Pull request by vasharma05**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://zeroheight.com/23a080e38/p/880723-introduction).
- [x] I checked for feature overlap with [**existing widgets**](https://om.rs/directory).

## Summary
This PR upgrades the i18next related packages. The changes in the new upgrades are as follows:

### Changes in i18next from v19 to v23

1. JSON V4 Support: https://www.i18next.com/misc/json-format#i-18-next-json-v4
2. Pluralization support: https://www.i18next.com/translation-function/plurals

### Changes in react-i18next 
No major changes as per documentation.

### i18next-parser from v5 → v6

1. Node v ≤ 14 not supported anymore
2. Deprecated `skipDefaultValues` and `useKeysAsDefaultValues` options

## Screenshots

*None.*

## Related Issue

*None.*

## Other

*None.*

### CODE REVIEW

File: packages/esm-service-queues-app/translations/am.json [hunk 1]
```diff
@@ -181,8 +181,8 @@
   "queueScreen": "Queue screen",
   "queueService": "Queue service",
   "queueStatus": "Queue status",
-  "recentScheduledVisits": "{count} visit(s) scheduled for +/- 7 days",
-  "recentScheduledVisits_plural": "{count} visit(s) scheduled for +/- 7 days",
+  "recentScheduledVisits_one": "{count} visit(s) scheduled for +/- 7 days",
+  "recentScheduledVisits_other": "{count} visit(s) scheduled for +/- 7 days",
```
Comment (jayasanka-sack) [inline packages/esm-service-queues-app/translations/am.json:185]:
```suggestion
  "recentScheduledVisits_one": "{count} visit scheduled for +/- 7 days",
  "recentScheduledVisits_other": "{count} visits scheduled for +/- 7 days",
```

Comment (jayasanka-sack) [reply packages/esm-service-queues-app/translations/am.json:185]: Applies for other translation files.

File: packages/esm-patient-search-app/translations/en.json [hunk 2]
```diff
@@ -34,13 +34,11 @@
   "searchingText": "Searching...",
   "searchPatient": "Search Patient",
   "searchResults": "Search Results",
-  "searchResultsCount": "{count} search result{plural}",
-  "searchResultsCount_plural": "{count} search result{plural}",
+  "searchResultsCount_one": "{count} search result",
+  "searchResultsCount_other": "{count} search result",
```
Comment (jayasanka-sack) [inline packages/esm-patient-search-app/translations/en.json:38]:
```suggestion
  "searchResultsCount_one": "{count} search result",
  "searchResultsCount_other": "{count} search results",
```

File: packages/esm-patient-search-app/translations/es.json [hunk 3]
```diff
@@ -34,13 +35,12 @@
   "searchingText": "Buscando...",
   "searchPatient": "Buscar Paciente",
   "searchResults": "Resultados de la Búsqueda",
-  "searchResultsCount": "{count} resultado{plural} de búsqueda",
-  "searchResultsCount_plural": "{count} resultados{plural} de búsqueda",
+  "searchResultsCount_one": "{count} resultado de búsqueda",
+  "searchResultsCount_many": "{count} resultado de búsqueda",
```
Comment (jayasanka-sack) [inline packages/esm-patient-search-app/translations/es.json:39]:
```suggestion
  "searchResultsCount_many": "{count} resultados de búsqueda",
```

**Review by jayasanka-sack — COMMENTED** [review]
Hey Vineet, thanks for your PR. I've made a few minor suggestions that I believe are typos. 

By the way, I couldn't help but notice that we're using single curly braces here for placeholders. As far as I know, the convention followed by standard i18n libraries uses double curly braces.

**Comment by vasharma05** [comment]
Thanks for the suggestions @jayasanka-sack !

Currently, we are using single braces app-wide. There's no hard binding on using double braces, but we'll surely get back on that too.

**Review by jayasanka-sack — APPROVED** [review]
Looks good to me. Thanks, Vineet!

**Pull request merged**